src/unix/stream.c

```diff
@@ -839,11 +840,36 @@ static void uv__write(uv_stream_t* stream) {
 #endif
   } else {
     do {
+      /* On some platforms, notably OSX, the sum of iov_len must not
+       * overflow a 32-bit integer.
+       */
+      size_t old_iov_len = iov[iovcnt - 1].iov_len;
```

Comment:
I admit to not understanding why it's necessary to save and restore this.  Why just the last iovec, why not all?